bundles/org.openhab.voice.voskstt/src/main/java/org/openhab/voice/voskstt/internal/VoskSTTService.java

```diff
@@ -0,0 +1,226 @@
+                while (keepStreaming.get()) {
+                    nbytes = audioStream.read(b);
+                    if (!keepStreaming.get()) {
+                        break;
+                    }
+                    if (isExpiredInterval(maxTranscriptionMillis, startTime)) {
+                        logger.debug("Stops listening, max transcription time reached");
+                        break;
+                    }
+                    if (!config.singleUtteranceMode && isExpiredInterval(maxSilenceMillis, lastInputTime)) {
+                        logger.debug("Stops listening, max silence time reached");
+                        break;
+                    }
+                    if (nbytes == 0) {
+                        trySleep(100);
+                        continue;
+                    }
+                    if (recognizer.acceptWaveForm(b, nbytes)) {
+                        lastInputTime = System.currentTimeMillis();
+                        var result = recognizer.getResult();
+                        logger.debug("Result: {}", result);
+                        ObjectMapper mapper = new ObjectMapper();
+                        var json = mapper.readTree(result);
+                        transcriptBuilder.append(json.get("text").asText()).append(" ");
+                        if (config.singleUtteranceMode) {
+                            break;
+                        }
+                    } else {
+                        logger.debug("Partial: {}", recognizer.getPartialResult());
+                    }
+                }
+                sttListener.sttEventReceived(new RecognitionStopEvent());
+                var transcript = transcriptBuilder.toString().trim();
+                logger.debug("Final: {}", transcript);
+                if (!transcript.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionEvent(transcript, 1F));
+                } else {
+                    if (!config.noResultsMessage.isBlank()) {
+                        sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.noResultsMessage));
+                    } else {
+                        sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("No results"));
+                    }
+                }
+            } catch (IOException e) {
+                logger.warn("Error running speech to text: {}", e.getMessage());
+                if (config.errorMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("Error"));
+                } else if (!config.errorMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.errorMessage));
+                }
+            }
+            try {
+                audioStream.close();
+            } catch (IOException e) {
+                logger.warn("IOException on close: {}", e.getMessage());
+            }
```

Comment:
I have a doubt: this is the STT service that closes the source ?
You did like that in the two other STT services ?

Reply:
No, but the Watson STT already does this internally, and for the Google one I will add a PR to do it. The current dialog processor implementation only ensure the stream is closed before open a new one which I think is fine, but due to that I think the STT service should be on charge of closing the stream when it will no longer be used to free the resources as soon as posible.

Reply:
Does it makes sense to you?

Reply:
You're tight. 
I am just asking myself if this is not something to fix rather in the dialog processor? I am not sure. 
If you really think it is more the job of each STT service to close the audio source, that's ok for me but it should be documented (probably in the interface description). 
WDYT?

Reply:
In my opinion the current implementation of the dialog processor is ok, as the stream will be always closed at the end. I think is something that doesn't need an interface change, maybe a comment on the interface method to tell that is ok to close the stream when finished is enough.

Related to this, there is something I want to change on all the voice add-ons (the ones I uploaded recently), I want stop closing the scheduled task inside the abort call, and just let it finish ensuring it does not produce more events. That close call was causing this error https://github.com/openhab/openhab-core/pull/2761.
Is it ok to you?

Reply:
Ok